src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -41,7 +41,7 @@ public void Init(Func<PushContext, Task> pipe, PushSettings settings)
             inputQueue = new MessageQueue(inputAddress.FullPath, false, true, QueueAccessMode.Receive);
             errorQueue = new MessageQueue(errorAddress.FullPath, false, true, QueueAccessMode.Send);
 
-            if (settings.TransactionSettings.IsTransactional && !QueueIsTransactional())
+            if (settings.RequiredConsistency.GetType() != typeof(AtMostOnce) && !QueueIsTransactional())
```

Comment:
I am not a big fan of using types in `if` statements but I accept this an an intermediate step. I believe the end goal might be to have a pump implementation per consistency mode. For now :ok: